stl/inc/__msvc_string_view.hpp

```diff
@@ -221,7 +221,15 @@ struct _WChar_traits : private _Char_traits<_Elem, unsigned short> {
         // compare [_First1, _First1 + _Count) with [_First2, ...)
 #if _HAS_CXX17
         if constexpr (is_same_v<_Elem, wchar_t>) {
+#if _HAS_CXX20
+            if (_STD is_constant_evaluated()) {
+                return __builtin_wmemcmp(_First1, _First2, _Count);
+            } else {
+                return _CSTD wmemcmp(_First1, _First2, _Count);
+            }
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
             return __builtin_wmemcmp(_First1, _First2, _Count);
+#endif // ^^^ !_HAS_CXX20 ^^^
         } else {
             return _Primary_char_traits::compare(_First1, _First2, _Count);
         }

```

Comment:
Finally, `length()` immediately below is the only function here that isn't following the "fallthrough to UCRT" pattern. Looking at `C:\Program Files (x86)\Windows Kits\10\Source\10.0.22621.0\ucrt\string\strnlen.cpp`, it appears that `wcslen()` has a SIMD-enlightened implementation, so we should use the same pattern.

Reply:
I didn't bother benchmarking or inspecting codegen for `__builtin_wcslen` (I am a lazy kitty :cat2:), but `_Primary_char_traits::length` presumably isn't going to do anything glorious for `char16_t`. Following the pattern uniformly in wide `char_traits` is simpler, even if this builtin is more intelligent than the others.